crates/client/src/client/async_client.rs

```diff
@@ -143,7 +143,7 @@ impl Clone for AsyncClient {
     fn clone(&self) -> Self {
         AsyncClient {
             exchange: self.exchange.clone(),
-            use_edns: true,
+            use_edns: self.use_edns,
```

Comment:
Hmm, if this seems right, can we just derive the `impl`?

Reply:
That's actually a really good point in favor of me just having misunderstood the intention of the code. This was my first read-through of this code. My understanding would be that if someone had configured edns one way or the other, it should be preserved on clone, but there might be a good reason to enable it on newly cloned instances.

If my suggested change is indeed correct, I'm happy to redo this PR with a derived impl. I'll wait for someone more knowledgeable to chime in.

Reply:
Not sure why this wasn’t a derive, but definitely seems like it should be. Was derive default possibly added to Rust after this? If the PR that introduced this doesn’t discuss this specifically, then I agree that it’s a mistake.

Reply:
I went and reviewed #1492 again and couldn't see a reason for it to have the special behavior. I've re-published the change to just replace the implementation with a derive.